jme3-core/src/main/java/com/jme3/anim/TransformTrack.java

```diff
@@ -189,12 +197,16 @@ public void setKeyframesRotation(Quaternion[] rotations) {
 
 
     /**
-     * Set the translations, rotations and scales for this bone track
+     * Set the translations, rotations, and/or scales
```

Comment:
Should we document that : 
1) the arrays translations, rotations, scales, times have to be of the same size. (strict note should be provided for this)
2) the net result is a `Transformation Matrix of (L x 3)` ; where L is the length of the times for clarifying how the api organize and arrange the inputs before processing the animation.

Reply:
I agree with 1 but not 2. The result isn't a "matrix" because the rotations are quaternions, not vectors.

Reply:
Well, that makes sense, could we correct that to a multidimensional matrix or just say that each row in this L x 3 is a frame that uses a corresponding time...?

Reply:
The problem is the first translation, the first rotation, the first scale and the first time should not be discrete, they should be enclosed in an entity called Frame or be documented that they represent a Frame.

Reply:
> could we correct that to a multidimensional array or just say that each row in this L x 3 is a frame
No, that would be misleading, because "array" and "matrix" both imply that all elements are of the same type. They aren't of the same type because the translations are vectors and the rotations are quaternions.

The class javadoc describes the track as "a list of transforms and times for keyframes" and the method javadoc describes each argument as being "for each frame". Short of creating a "Keyframe" class and rewriting the entire module, I don't see how to make this any clearer.